### PR TITLE
AD: Add ChainRules extension for MTKParameters construction

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

Needed for https://github.com/SciML/ModelingToolkitNeuralNets.jl/issues/20

Basically, adds a simple `rrule` to make the MTKParameters constructor work out of the box. Currently only handles tunables

cc @AayushSabharwal

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Add a quick test?

**Comment by DhairyaLGandhi** [comment]
Indeed, I am working on one.

**Comment by ChrisRackauckas** [comment]
I'll merge with the depwarn.
